api/codelists.go

```diff
@@ -15,22 +15,38 @@ import (
 func (c *CodeListAPI) getCodeLists(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	filterBy := r.URL.Query().Get("type")
+	logData := log.Data{}
+	offsetParameter := r.URL.Query().Get("offset")
+	limitParameter := r.URL.Query().Get("limit")
+	offset := c.defaultOffset
+	limit := c.defaultLimit
+	var err error
 
-	// get limit from query parameters, or default value
-	limit, err := GetPositiveIntQueryParameter(r.URL.Query(), "limit", c.defaultLimit)
-	if err != nil {
-		handleError(ctx, "failed to obtain a positive integer value for limit query parameter", log.Data{"limit": limit}, err, w)
-		return
+	log.Event(ctx, "attempting to get code lists", log.INFO, log.Data{"type": filterBy})
+
+	if offsetParameter != "" {
+		logData["offset"] = offsetParameter
+		offset, err = ValidatePositiveInt(offsetParameter)
+		if err != nil {
+			handleError(ctx, "failed to obtain a positive integer value for offset query parameter", log.Data{}, err, w)
```

Comment:
handleError currently has options for ErrNotFound or internalServerError. Just wondering ..
In this case if it fails, are we ok to return an internal server error? or do we want http.StatusBadRequest?

Reply:
I'm not too sure on this one to be honest. This PR is just an update of a PR of mine that was approved by Carl a few days ago - https://github.com/ONSdigital/dp-code-list-api/pull/104/files

The error handling hasn't changed and Carl didn't comment on it previously so I think this is probably ok

Reply:
Happy to look into it further though if you think otherwise